lib/libsys/mmap_c18n.c

```diff
@@ -15,18 +15,25 @@
 
 #include "libc_private.h"
 
+__weak_reference(__mmap, mmap);
+__weak_reference(__mmap, _mmap);
+
 void *
-mmap(void *addr, size_t len, int prot, int flags, int fd, off_t offset)
+__mmap(void *addr, size_t len, int prot, int flags, int fd, off_t offset)
```

Comment:
Where are the prototypes for these?

Reply:
Hm, it seems symbols like __openat don't have Symbol.map entries. @brooksdavis do you know what those symbols are for? The default assembly implementations don't provide them.

Reply:
The double underscore syscall aliases generally seem to be blind copy and paste nonsense (I believe that for example the weak __openat symbol bit doesn't even work because __sys_openat isn't in scope).  I've got a patch to delete all of them around somewhere.

Reply:
Those work because it's done in inline assembly, though __openat does end up as STB_LOCAL rather than STB_GLOBAL in the linked output. Not quite sure how.

Reply:
If there intent is that __mmap be called by something that overrides `mmap` I'd probably name it __libsys_mmap.

Reply:
@jrtc27 I'm getting the following if I make all these stubs `static`.
```
ld.lld: error: version script assignment of 'FBSD_1.0' to symbol 'rfork' failed: symbol not defined
ld.lld: error: version script assignment of 'FBSD_1.0' to symbol 'thr_exit' failed: symbol not defined
ld.lld: error: version script assignment of 'FBSD_1.0' to symbol 'vfork' failed: symbol not defined
ld.lld: error: version script assignment of 'FBSD_1.0' to symbol 'mmap' failed: symbol not defined
```

@brooksdavis The intent of these double-underscored functions is to provide a common c18n-aware implementation for both `_<syscall>` and `__sys_<syscall>`. Right now I add them to `PSEUDO` in `Makefile.inc` but I think they should be added to `NO_UNDERSCORE` instead. Is that right?

Reply:
You probably do want NO_UNDERSCORE if you want to provide the _mmap implementations. That being said, I don't think you need to provide _mmap and can leave it as a weak alias for __sys_mmap for now. I plan to greatly reduce the number of _<syscall> symbols in the not too distant future and maybe even kill them off entirely as their purpose was to support libc_r, an obsolete threading mechanism that provided no concurrency.